src/promptflow/promptflow/batch/_batch_inputs_processor.py

```diff
@@ -0,0 +1,296 @@
+                    "List lengths are: {all_lengths_without_line_number}. "
+                    "Please make sure these lists have the same length or add 'line_number' key to each dictionary."
+                ),
+                all_lengths_without_line_number=all_lengths_without_line_number,
+            )
+
+        # Collect each line item from each input.
+        tmp_dict = {}
+        for input_key, list_of_one_input in input_dict.items():
+            if input_key in all_lengths_without_line_number:
+                # Assume line_number start from 0.
+                for index, one_line_item in enumerate(list_of_one_input):
+                    if index not in tmp_dict:
+                        tmp_dict[index] = {}
+                    tmp_dict[index][input_key] = one_line_item
+            else:
+                for one_line_item in list_of_one_input:
+                    if LINE_NUMBER_KEY in one_line_item:
+                        index = one_line_item[LINE_NUMBER_KEY]
+                        if index not in tmp_dict:
+                            tmp_dict[index] = {}
+                        tmp_dict[index][input_key] = one_line_item
+        result = []
+        for line, values_for_one_line in tmp_dict.items():
+            # Missing input is not acceptable line.
+            if len(values_for_one_line) != len(input_dict):
+                continue
+            values_for_one_line[LINE_NUMBER_KEY] = line
+            result.append(values_for_one_line)
+        return result
+
+
+def apply_inputs_mapping(
+    inputs: Mapping[str, Mapping[str, Any]],
+    inputs_mapping: Mapping[str, str],
+) -> Dict[str, Any]:
+    """Apply input mapping to inputs for new contract.
+
+    .. admonition:: Examples
+
+        .. code-block:: python
+
+            inputs: {
+                "data": {"answer": "I'm fine, thank you.", "question": "How are you?"},
+                "baseline": {"answer": "The weather is good."},
+            }
+            inputs_mapping: {
+                "question": "${data.question}",
+                "groundtruth": "${data.answer}",
+                "baseline": "${baseline.answer}",
+                "deployment_name": "literal_value",
+            }
+
+            Returns: {
+                "question": "How are you?",
+                "groundtruth": "I'm fine, thank you."
+                "baseline": "The weather is good.",
+                "deployment_name": "literal_value",
+            }
+
+    :param inputs: A mapping of input keys to their corresponding values.
+    :type inputs: Mapping[str, Mapping[str, Any]]
+    :param inputs_mapping: A mapping of input keys to their corresponding mapping expressions.
+    :type inputs_mapping: Mapping[str, str]
+    :return: A dictionary of input keys to their corresponding mapped values.
+    :rtype: Dict[str, Any]
+    :raises InputMappingError: If any of the input mapping relations are not found in the inputs.
+    """
+    import re
```

Comment:
Why is this not in file top?

Reply:
Just copy from `flow_executor`. I'll update it.